torchgeo/datasets/spacenet1.py

```diff
@@ -0,0 +1,215 @@
+"""Spacenet1 dataset."""
+
+import glob
+import os
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import fiona
+import numpy as np
+import rasterio as rio
+import torch
+from affine import Affine
+from rasterio.crs import CRS
+from rasterio.features import rasterize
+from torch import Tensor
+
+from torchgeo.datasets.geo import RasterDataset
+from torchgeo.datasets.utils import (
+    check_integrity,
+    download_radiant_mlhub,
+    extract_archive,
+)
+
+
+class Spacenet1(RasterDataset):
+    """Spacenet 1: Building Detection v1 Dataset
+
+    `Spacenet 1 <https://spacenet.ai/spacenet-buildings-dataset-v1/>`_
+    is a dataset of building footprints over the city of Rio de Janeiro.
+
+    Dataset features:
+
+    * No. of images - 6940 (8 Band) + 6940 (RGB)
+    * No. of polygons - 382,534 building labels
+    * Area Coverage - 2544 sq km
+
+    Dataset format:
+
+    * Imagery - Raw 8 band Worldview-3 (GeoTIFF) & Pansharpened RGB image (GeoTIFF)
+    * Labels - GeoJSON
+
+    If you are using data from SpaceNet in a paper, please use the following citation:
+
+    * Van Etten, A., Lindenbaum, D., & Bacastow, T.M. (2018). SpaceNet: A Remote
+      Sensing Dataset and Challenge Series. ArXiv, abs/1807.01232.
+
+    .. note::
+
+       This dataset requires the following additional library to be installed:
+
+       * `radiant-mlhub <https://pypi.org/project/radiant-mlhub/>`_ to download the
+         imagery and labels from the Radiant Earth MLHub
+
+    """
+
+    dataset_id = "spacenet1"
+    md5 = "e6ea35331636fa0c036c04b3d1cbf226"
+    filename_glob = "RGB.tif"  # To prevent reading .tif.aux.xml
+    raw_8band_glob = "8Band.tif"
+    label_glob = "labels.geojson"
+    foldername = "sn1_AOI_1_RIO"
```

Comment:
I would rather not place these in a subfolder of `root`. It prevents users from using existing datasets in the cloud unless the directory structure matches exactly.

Reply:
The subfolder is due to how the spacenet1 dataset is packaged. When I download the spacenet1 dataset via the radiant earth mlhub api and extract it, it creates a folder named `sn1_AOI_1_RIO` as follows.

```
sn1_AOI_1_RIO
├── sn1_AOI_1_RIO_img1
│   ├── 8Band.tif
│   ├── RGB.tif
│   └── stac.json
├── sn1_AOI_1_RIO_img1-labels
│   ├── labels.geojson
│   └── stac.json
......
```

I could use `shutil` to move all the contents of `sn1_AOI_1_RIO` to root. Is that preferable?

Reply:
Ah, I see. In that case it's fine as is.